src/client/base/tokio.rs

```diff
@@ -59,11 +59,11 @@ impl TokioClient {
         Self { client: hyper::Client::builder().build(connector::create()) }
     }
 
-    pub fn execute<T: DeserializeOwned + Send + 'static>(
-        &self,
+    pub fn execute<'a, T: DeserializeOwned + Send + 'a>(
+        &'a self,
         request: Request,
-        strategy: &RequestStrategy,
-    ) -> Response<T> {
+        strategy: &'a RequestStrategy,
+    ) -> Response<'a, T> {
```

Comment:
I'm a little confused by the lifetimes here - why is tying the `Response` lifetime tied to the `client` lifetime? Looks like in the function body the `client` gets cloned anyway, so could that `client` just be moved into the future so that no lifetime is needed?

Reply:
You are probably correct in questioning this. The PR is a little sloppy at the moment as it took a fair bit of experimentation.